Source/Result.swift

```diff
@@ -100,3 +100,82 @@ extension Result: CustomDebugStringConvertible {
         }
     }
 }
+
+// MARK: - Functional Tools
+
+extension Result {
+    /// Returns the success value, or throws the failure error.
+    ///
+    ///     let possibleString: Result<String> = .success("success")
+    ///     try print(possibleString.unwrap())
+    ///     // Prints "success"
+    ///
+    ///     let noString: Result<String> = .failure(error)
+    ///     try print(noString.unwrap())
+    ///     // Throws error
+    func unwrap() throws -> Value {
```

Comment:
The (unfinished) proposal by @erica is https://gist.github.com/erica/3daa8ec77aef2feaefca3a3a19aedee3, introduced in this [thread](https://lists.swift.org/pipermail/swift-evolution/Week-of-Mon-20161114/028833.html) on swift-evolution mailing list.